sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/OrcQuerySuite.scala

```diff
@@ -91,6 +91,16 @@ class OrcQuerySuite extends QueryTest with BeforeAndAfterAll with OrcTest {
     }
   }
 
+  test("Read/write UserDefinedType") {
+    withTempPath { path =>
+      val data = Seq((1, new UDT.MyDenseVector(Array(0.25, 2.25, 4.25))))
+      val udtDF = data.toDF("id", "vectors")
+      udtDF.write.orc(path.getAbsolutePath)
+      val readBack = spark.read.schema(udtDF.schema).orc(path.getAbsolutePath)
```

Comment:
Curious : how does this work ? I mean you added support for `UserDefinedType` in `wrapper` side, but at the `unwrapper` side I don't see `UserDefinedType` being handled.